csharp/src/Ice/LocatorDiscovery/Locator.cs

```diff
@@ -254,6 +210,98 @@ internal void FoundLocator(ILocatorPrx locator)
             }
         }
 
+        private Locator(Communicator communicator)
+        {
+            const string defaultIPv4Endpoint = "udp -h 239.255.0.1 -p 4061";
+            const string defaultIPv6Endpoint = "udp -h \"ff15::1\" -p 4061";
+
+            string? lookupEndpoints = communicator.GetProperty("Ice.LocatorDiscovery.Lookup");
+            if (lookupEndpoints == null)
+            {
+                List<string> endpoints = new ();
+                List<string> ipv4Interfaces = Network.GetInterfacesForMulticast("0.0.0.0", Network.EnableIPv4);
+                List<string> ipv6Interfaces = Network.GetInterfacesForMulticast("::0", Network.EnableIPv6);
+
+                endpoints.AddRange(ipv4Interfaces.Select(i => $"{defaultIPv4Endpoint} --interface \"{i}\""));
+                endpoints.AddRange(ipv6Interfaces.Select(i => $"{defaultIPv6Endpoint} --interface \"{i}\""));
+
+                lookupEndpoints = string.Join(":", endpoints);
+            }
+
+            _timeout = communicator.GetPropertyAsTimeSpan("Ice.LocatorDiscovery.Timeout") ??
+                 TimeSpan.FromMilliseconds(300);
+            if (_timeout == System.Threading.Timeout.InfiniteTimeSpan)
+            {
+                _timeout = TimeSpan.FromMilliseconds(300);
+            }
+
+            _lookup = ILookupPrx.Parse($"IceLocatorDiscovery/Lookup -d:{lookupEndpoints}", communicator).Clone(
+                    clearRouter: true,
+                    invocationTimeout: _timeout);
+
+            if (communicator.GetProperty("Ice.LocatorDiscovery.Reply.Endpoints") == null)
+            {
+                communicator.SetProperty("Ice.LocatorDiscovery.Reply.Endpoints", "udp -h \"::0\" -p 0");
+            }
+            communicator.SetProperty("Ice.LocatorDiscovery.Reply.ProxyOptions", "-d");
+
+            if (communicator.GetProperty("Ice.LocatorDiscovery.Locator.Endpoints") == null)
```

Comment:
Same here, shouldn't we change this adapter to be a local only adapter?

Reply:
Yes, fixed.